packages/ckeditor5-html-support/src/converters.js

```diff
@@ -89,7 +89,17 @@ export function createObjectView( viewName, modelElement, writer ) {
 export function viewToAttributeInlineConverter( { view: viewName, model: attributeKey }, dataFilter ) {
 	return dispatcher => {
 		dispatcher.on( `element:${ viewName }`, ( evt, data, conversionApi ) => {
-			const viewAttributes = dataFilter._consumeAllowedAttributes( data.viewItem, conversionApi );
+			let viewAttributes = dataFilter._consumeAllowedAttributes( data.viewItem, conversionApi );
+
+			// Do not apply an attribute if the element itself is already consumed and there is no view attributes to store.
+			if ( !viewAttributes && !conversionApi.consumable.test( data.viewItem, { name: true } ) ) {
```

Comment:
It seems to me that the fragment from line 95 to 107 can be simplified to:

```
if ( !viewAttributes ) {
	conversionApi.consumable.consume( data.viewItem, { name: true } );
	
	return true;
}
```

Or am I missing something?

If there are no attributes, the converter makes sure that the element is consumed and that's it.

Reply:
There is a case of an element that does not have any attributes but has to be converted to an empty attribute object to be preserved. So for example `<cite>` must not only be consumed but also converted to `htmlCite={}`.

Reply:
Ugh, I just noticed that there is a merge error.

Reply:
Yes, there is the same `if` twice there.

Reply:
Okay, now it probably makes more sense.